src/android/LocalyticsPlugin.java

```diff
@@ -1076,11 +1104,14 @@ public void run() {
         } else if (action.equals("setCallToActionListener")) {
             ctaListener = new CDCTAListener(callbackContext);
             Localytics.setCallToActionListener(ctaListener);
+            callbackContext.success();
             return true;
         } else if (action.equals("removeCallToActionListener")) {
             ctaListener = null;
             Localytics.setCallToActionListener(null);
-            callbackContext.success();
+            PluginResult result = new PluginResult(PluginResult.Status.OK);
+            result.setKeepCallback(true);
```

Comment:
I think we need to keep in setCallToActionListener not here. Here we should not keep it.

Reply:
whoops ya nice catch